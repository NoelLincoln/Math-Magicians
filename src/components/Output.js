import { PropTypes } from 'prop-types';

const Output = ({ value }) => (
  <div className="result">
    <input type="text" value={value} id="inputField" />
  </div>
);

Output.propTypes = {
  value: PropTypes.string.isRequired,
};

export default Output;
