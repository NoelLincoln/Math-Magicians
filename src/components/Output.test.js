import React from 'react';
import { render } from '@testing-library/react';
import Output from './Output';

test('Output component snapshot', () => {
  const { asFragment } = render(<Output value="Sample Value" />);
  expect(asFragment()).toMatchSnapshot();
});

test('Output component displays the value correctly', () => {
  const { getByTestId } = render(<Output value="12345" />);
  const inputField = getByTestId('inputField');

  expect(inputField.value).toBe('12345');
});
