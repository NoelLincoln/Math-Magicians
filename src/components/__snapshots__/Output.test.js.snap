// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output component snapshot 1`] = `
<DocumentFragment>
  <div
    class="result"
  >
    <input
      data-testid="inputField"
      readonly=""
      type="text"
      value="Sample Value"
    />
  </div>
</DocumentFragment>
`;
